### DISCUSSION TITLE
Build Failed, Command CodeSign failed with a nonzero exit code

### DISCUSSION THREAD

**Discussion by LaKing** [ExistentialAudio/BlackHole#718] [Q&A]
Dear Developers, ...

I'm rarely compiling stuff, but need to compile Blackhole now.

I did as described, but can't get past the error `Command CodeSign failed with a nonzero exit code`.

There are some additional warnings, 
```
BlackHole.c:436:31: A function declaration without a prototype is deprecated in all versions of C
...
BlackHole.c:441:40: A function declaration without a prototype is deprecated in all versions of C
```

I'm logged in with my Apple ID, downloaded the git repo, and whatever I set in Signing & Capabilities, I'm stuck.

Could someone shed some light on this, what needs to be set? Thank you.

**Comment by devinroth** [comment]
There’s a script in build phases. Try disabling that. I think that’s caused some issues for me in the past.

Devin

**Reply by LaKing** [reply]
Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

This is probably what you mean. This was not a show-stopper though.

After starting over 10 times and trying 100 variations I finally got it compiled. I have no idea what was wrong, or how I fixed it.

The function declaration warning is an issue I think, ... I should open one I guess.

**Comment by devinroth** [comment]
The function declaration is just a warning. I highly doubt it’s the cause of your issues. I’m off for the next few days for Easter. I’ll take a closer look when I get back. That being said, I’ve built this code many many times without any problems. My guess is this might be an issue with the latest version of Xcode.

Devin
